src/transformer/events/core/message_viewed.php

```diff
@@ -0,0 +1,78 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Transformer fn for message viewed event
+ *
+ * @package   logstore_xapi
+ * @copyright Daniel Bell <daniel@yetanalytics.com>
+ *            
+ * @license   https://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace src\transformer\events\core;
+
+use src\transformer\utils as utils;
+use src\transformer\utils\get_activity as activity;
+
+/**
+ * Transformer fn for message viewed event
+ *
+ * @param array $config The transformer config settings.
+ * @param \stdClass $event The event to be transformed.
+ * @return array
+ */
+
+function message_viewed(array $config, \stdClass $event) {
+    global $CFG;
+    $repo = $config['repo'];
+    if (isset($event->objecttable) && isset($event->objectid)) {
+        $event_object = $repo->read_record_by_id($event->objecttable, $event->objectid);
+    } else {
+        $event_object = array();
+    }
+
+    $user=$repo->read_record_by_id('user',$event->userid); 
+    $course = (isset($event->courseid) && $event->courseid != 0) ? $repo->read_record_by_id('course', $event->courseid) : null;
+    $lang = utils\get_course_lang(($course ? $course :  $repo->read_record_by_id("course",1)));
+
+    $statement = [
+        'actor' => utils\get_user($config,$user),
+        'verb' => ['id' => 'http://id.tincanapi.com/verb/viewed',
+                   'display' =>  ['en' => 'Viewed']
+        ],
+        'object' => [
+            'id' =>  $config['app_url'].'/course/view.php?id='.$event->objectid,
+            'definition' => [
+                'type' =>  "http://id.tincanapi.com/activitytype/chat-message",
+                'name' => [$lang => $event_object->subject ?? 'no subject'],
+                'description' => [$lang => $event_object->smallmessage],
```

Comment:
Just curious, does the message used here have HTML tags? I've noticed this in a few other features and was curious. If so we might consider stripping them out with one of the existing utils in the project.

Reply:
Just to be clear, I mean when the event is fired, I can see in the test data there are no tags.

Reply:
i'll have to test

Reply:
For posterity, yes they were HTML and @invaliduser stripped them.